src/rgw/rgw_rados.cc

```diff
@@ -426,7 +426,7 @@ void RGWZoneGroup::post_process_params()
   for (map<string, RGWZone>::iterator iter = zones.begin(); iter != zones.end(); ++iter) {
     RGWZone& zone = iter->second;
     zone.log_data = log_data;
-    zone.log_meta = (is_master && zone.id == master_zone);
+    zone.log_meta = (is_master && zone.id == master_zone && zone.name != default_zone_name);
```

Comment:
Also post_process_params is called when updating placement group, I think we should update log_meta only if the zone was set to master and not touch it in other cases